contracts/utils/contracts/src/MixinSafeMathRichErrors.sol

```diff
@@ -0,0 +1,47 @@
+pragma solidity ^0.5.9;
+
+import "./RichErrors.sol";
+
+
+contract MixinSafeMathRichErrors is
+    RichErrors
+{
+    // bytes4(keccak256("Uint256OverflowError(uint256,uint256)"))
+    bytes4 internal constant UINT256_OVERFLOW_ERROR =
+        0x55101607;
+
+    // bytes4(keccak256("Uint256UnderflowError(uint256,uint256)"))
+    bytes4 internal constant UINT256_UNDERFLOW_ERROR =
+        0x60ee612f;
+
+    // solhint-disable func-name-mixedcase
+    function Uint256OverflowError(
```

Comment:
These SafeMath errors don't differentiate between addition and multiplication. What if we only had a single `SafeMathError` that took a `SafeMathErrorCode` in addition to `a` and `b`?

```
enum SafeMathErrorCodes {
    UINT256_MULTIPLICATION_OVERFLOW,
    UINT256_ADDITION_OVERFLOW,
    UINT256_SUBTRACTION_UNDERFLOW
}

function SafeMathError(
    SafeMathErrorCodes errorCode,
    uint256 a,
    uint256 b
)
````

Reply:
That makes sense to me. I'll update that.